pytensor/tensor/rewriting/linalg.py

```diff
@@ -60,11 +60,36 @@
     solve_triangular,
 )
 
+from pytensor.tensor.slinalg import BlockDiagonal
```

Comment:
Make sure you have `pre-commit` and you've done `pre-commit install` in your dev environment. You have doubled imports and other issues this tool with help you check.